iris-mpc-cpu/src/execution/hawk_main/rot.rs

```diff
@@ -0,0 +1,77 @@
+use std::ops::Deref;
+
+use iris_mpc_common::ROTATIONS;
+use itertools::Itertools;
+
+/// VecRots are lists of things for each rotation.
+pub struct VecRots<R> {
+    pub rotations: Vec<R>,
+}
+
+impl<R> Deref for VecRots<R> {
+    type Target = Vec<R>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.rotations
+    }
+}
+
+impl<R> VecRots<R> {
+    pub fn new(rotations: Vec<R>) -> Self {
+        assert_eq!(rotations.len(), ROTATIONS);
+        Self { rotations }
+    }
+
+    /// Get the item attached to the center rotation.
+    pub fn center(&self) -> &R {
+        &self.rotations[self.rotations.len() / 2]
+    }
+
+    /// Get the item attached to the center rotation.
+    pub fn into_center(self) -> R {
+        let middle = self.rotations.len() / 2;
+        self.rotations.into_iter().nth(middle).unwrap()
```

Comment:
out of curiosity, why not `self.rotations[middle].clone()`?

Reply:
this way saves a `clone` call, which can be relevant for expensive to clone `R`s I guess. In practice I agree it should probably not matter.